org-node/src/client.rs

```diff
@@ -353,20 +353,32 @@ impl Client {
 
                     match Client::get_project_head(&urn, &api).await {
                         Ok(Some(Head { remote, branch })) => {
-                            if let Err(err) = Client::set_head(&urn, &remote, &branch, &paths) {
-                                tracing::error!(target: "org-node", "Error setting head: {}", err);
-                                ack.send(Err(err)).ok();
-                                continue;
-                            }
-
-                            if let Err(err) = Client::sign_refs(urn, &api).await {
-                                tracing::error!(target: "org-node", "Error signing refs: {}", err);
-                                ack.send(Err(err)).ok();
-                                continue;
+                            match Client::set_head(&urn, &remote, &branch, &paths) {
+                                Err(err) => {
+                                    tracing::error!(target: "org-node", "Error setting head: {}", err);
+                                    ack.send(Err(err)).ok();
+                                    continue;
+                                }
+                                Ok(rev) => {
+                                    if let Err(err) = Client::sign_refs(urn.clone(), &api).await {
+                                        tracing::error!(target: "org-node", "Error signing refs: {}", err);
+                                        ack.send(Err(err)).ok();
+                                        continue;
+                                    }
+
+                                    // Announce the peer update to the network after setting the head.
+                                    if let Err(e) = api.announce(Payload {
+                                        urn,
+                                        rev: Some(rev),
+                                        origin: Some(peer_id),
+                                    }) {
+                                        tracing::error!(target: "org-node", "Error announcing ref update: {:?}", e);
```

Comment:
also using debug print here due to error type of `Gossip::Payload`, can log an oid/urn instead if desired.

Reply:
removed announce statement from protocol events as extraneous